documentation/ingestion_server/guides/migrate.md

```diff
@@ -0,0 +1,245 @@
+# Index migration runbook
+
+From time to time, we will need to update our Elasticsearch indices. These
+modifications can be classified into two broad-strokes categories, depending on
+whether the changes affect the main consumer of the indices, the API.
+
+## Migration types
+
+### API-free
+
+These changes are safe modifications to the ES schema that do not affect the
+API. As such they do not need any migration process. Examples:
+
+- addition of new fields or subfields
+- removal of fields that are not referenced or used by the API
+- changing the type to another compatible type (like `text` &harr; `keyword` )
```

Comment:
This case is API-free only if the API does not refer to the field using the `.keyword` title (as we used to have for source, for example).

Reply:
No that's a misunderstanding because here `.keyword` was referring to the inner field which was confusingly named `keyword` (and had the type of `keyword`). 

Changing the type of a field from `text` to `keyword` does not entail any API changes and is API-free. It only changes how that field is indexed and searched internally by ES.